modules/intentIqAnalyticsAdapter.js

```diff
@@ -0,0 +1,234 @@
+  eidsReadTime: 'eidt',
+  agentId: 'aid',
+  auctionEidsLength: 'aeidln',
+  wasServerCalled: 'wsrvcll',
+  referrer: 'vrref',
+  isInBrowserBlacklist: 'inbbl',
+  prebidVersion: 'pbjsver',
+  partnerId: 'partnerId'
+};
+
+let iiqAnalyticsAnalyticsAdapter = Object.assign(adapter({ defaultUrl, analyticsType }), {
+  initOptions: {
+    lsValueInitialized: false,
+    partner: null,
+    fpid: null,
+    currentGroup: null,
+    dataInLs: null,
+    eidl: null,
+    lsIdsInitialized: false,
+    manualReport: false
+  },
+  track({ eventType, args }) {
+    switch (eventType) {
+      case BID_WON:
+        bidWon(args);
+        break;
+      default:
+        break;
+    }
+  }
+});
+
+// Events needed
+const {
+  BID_WON
+} = EVENTS;
+
+function readData(key) {
+  try {
+    if (storage.hasLocalStorage()) {
+      return storage.getDataFromLocalStorage(key);
+    }
+    if (storage.cookiesAreEnabled()) {
+      return storage.getCookie(key);
+    }
+  } catch (error) {
+    logError(error);
+  }
+}
+
+function initLsValues() {
+  if (iiqAnalyticsAnalyticsAdapter.initOptions.lsValueInitialized) return;
+  iiqAnalyticsAnalyticsAdapter.initOptions.fpid = JSON.parse(readData(FIRST_PARTY_KEY));
+  let iiqArr = config.getConfig('userSync.userIds').filter(m => m.name == 'intentIqId');
+  if (iiqArr && iiqArr.length > 0) iiqAnalyticsAnalyticsAdapter.initOptions.lsValueInitialized = true;
+  if (!iiqArr) iiqArr = [];
+  if (iiqArr.length == 0) {
+    iiqArr.push({
+      'params': {
+        'partner': -1,
+        'group': 'U'
+      }
+    })
+  }
+  if (iiqArr && iiqArr.length > 0) {
+    if (iiqArr[0].params && iiqArr[0].params.partner && !isNaN(iiqArr[0].params.partner)) {
+      iiqAnalyticsAnalyticsAdapter.initOptions.partner = iiqArr[0].params.partner;
+      iiqAnalyticsAnalyticsAdapter.initOptions.currentGroup = iiqAnalyticsAnalyticsAdapter.initOptions.fpid.group;
+    }
+  }
+}
+
+function initReadLsIds() {
+  if (isNaN(iiqAnalyticsAnalyticsAdapter.initOptions.partner) || iiqAnalyticsAnalyticsAdapter.initOptions.partner == -1) return;
+  try {
+    iiqAnalyticsAnalyticsAdapter.initOptions.dataInLs = null;
+    let iData = readData(FIRST_PARTY_DATA_KEY + '_' + iiqAnalyticsAnalyticsAdapter.initOptions.partner)
+    if (iData) {
+      iiqAnalyticsAnalyticsAdapter.initOptions.lsIdsInitialized = true;
+      let pData = JSON.parse(iData);
+      iiqAnalyticsAnalyticsAdapter.initOptions.dataInLs = pData.data;
+      iiqAnalyticsAnalyticsAdapter.initOptions.eidl = pData.eidl || -1;
+    }
+  } catch (e) {
+    logError(e)
+  }
+}
+
+function bidWon(args) {
+  if (!iiqAnalyticsAnalyticsAdapter.initOptions.lsValueInitialized) { initLsValues(); }
+  if (iiqAnalyticsAnalyticsAdapter.initOptions.lsValueInitialized && !iiqAnalyticsAnalyticsAdapter.initOptions.lsIdsInitialized) { initReadLsIds(); }
+  if (!iiqAnalyticsAnalyticsAdapter.initOptions.manualReport) {
+    ajax(constructFullUrl(preparePayload(args, true)), undefined, null, { method: 'GET' });
```

Comment:
just some friendly advice, your get requests are going to run out of characters and you'll eventually want to switch to post one day. the sooner you do, the less painful it will be for you

Reply:
also, if you pass options = keepalive, the page performance will be better and you'll be more likely to get the events after the page dies

Reply:
Thanks a lot for the suggestion, @patmmccann! Yes, we're planning to implement these in the next updates. We would like to get the adapter approved and merged first.